apps/studio/src/features/dashboard/components/ResourceTable/ResourceTableMenu.tsx

```diff
@@ -10,14 +11,24 @@ import {
 
 import type { ResourceTableData } from "./types"
 import { MenuItem } from "~/components/Menu"
+import { moveResourceAtom } from "~/features/editing-experience/atoms"
 
 interface ResourceTableMenuProps {
   title: ResourceTableData["title"]
   resourceId: ResourceTableData["id"]
   type: ResourceTableData["type"]
+  permalink: ResourceTableData["permalink"]
 }
 
-export const ResourceTableMenu = ({ title, type }: ResourceTableMenuProps) => {
+export const ResourceTableMenu = ({
+  resourceId,
+  title,
+  type,
+  permalink,
+}: ResourceTableMenuProps) => {
+  const setMoveResourceId = useSetAtom(moveResourceAtom)
+  const prepareMoveItem = () =>
+    setMoveResourceId({ resourceId, title, permalink })
```

Comment:
nit: should we rename this for clarity? 

```suggestion
    setMoveResource({ resourceId, title, permalink })
```

Reply:
good idea updated